src/iaux-underlined-tab-bar.ts

```diff
@@ -0,0 +1,195 @@
+import {
+  html,
+  css,
+  LitElement,
+  property,
+  customElement,
+  query,
+  queryAll,
+  PropertyValues,
+  internalProperty,
+  CSSResult,
+} from 'lit-element';
+import { nothing } from 'lit-html';
+import './iaux-underlined-tab-bar-loading-dots';
+
+export interface TopTab {
+  displayName: string;
+  tabTitle?: string;
+}
+
+@customElement('iaux-underlined-tab-bar')
+export class UnderlinedTabBar extends LitElement {
+  @property({ type: Array }) entries: TopTab[] = [];
+
+  @property({ type: Number }) selectedIndex = 0;
+
+  @property({ type: Number }) widthMultiplier = 0.8;
+
+  @property({ type: Boolean }) isLoading = false;
+
+  @internalProperty() animatedUnderline = false;
+
+  @query('.horizontalScroll') container!: HTMLDivElement;
+
+  @queryAll('button') listOfButtons!: HTMLButtonElement[];
+
+  private itemClicked(e: Event, index: number) {
+    const event = new CustomEvent('itemclicked', {
+      detail: {
+        index,
+      },
+    });
+    const indexDistance = Math.abs(this.selectedIndex - index);
+    const animationTime = Math.min(0.25 * indexDistance, 0.5);
+    this.style.setProperty(
+      '--tabBarUnderlineAnimationDuration',
+      `${animationTime}s`
+    );
+    this.selectedIndex = index;
+    this.dispatchEvent(event);
+    this.animatedUnderline = true;
+  }
+
+  private updateUnderline() {
+    const buttonSelected = this.listOfButtons[this.selectedIndex];
+    if (!buttonSelected) return;
+    const boundingRect = buttonSelected.getBoundingClientRect();
+    const containerRect = this.container.getBoundingClientRect();
+    const containerDiff = boundingRect.x - containerRect.x;
+
+    const lineWidth = boundingRect.width * this.widthMultiplier;
+
+    const centerLineMargin = (boundingRect.width - lineWidth) / 2;
+    const underlineLeft =
+      containerDiff + centerLineMargin + this.container.scrollLeft;
+    this.style.setProperty(
+      '--tabBarUnderlineLeftPosition',
+      `${underlineLeft}px`
+    );
+    this.style.setProperty('--tabBarUnderlineWidth', `${lineWidth}px`);
+  }
+
+  updated(changed: PropertyValues) {
+    if (changed.has('selectedIndex') || changed.has('widthMultiplier')) {
+      this.updateUnderline();
+    }
+  }
+
+  render() {
+    return html`
+      <div class="headD">
+        <div class="horizontalScroll">
+          <ul class="headding">
+            ${this.entries.map(
+              (entry, index) =>
+                html`
+                  <li>
+                    <button
+                      @click=${(e: Event) => this.itemClicked(e, index)}
+                      class=${index === this.selectedIndex ? 'active' : ''}
+                    >
+                      ${entry.displayName}
+                    </button>
+                  </li>
+                `
+            )}
+            ${this.isLoading
+              ? html` <li>
```

Comment:
could this `<li>` be moved to the next line so that it vertically aligns with its closing `</li>` below?